universe/assets_3d/api_assets_3d.go

```diff
@@ -0,0 +1,318 @@
+package assets_3d
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+
+	"github.com/momentum-xyz/ubercontroller/types/entry"
+	"github.com/momentum-xyz/ubercontroller/universe"
+)
+
+func (a *Assets3d) apiCreateAsset3d(c *gin.Context) {
+	in := struct {
+		Meta    *entry.Asset3dMeta    `json:"meta"`
+		Options *entry.Asset3dOptions `json:"options"`
+	}{}
+
+	if err := c.ShouldBindJSON(&in); err != nil {
+		err = errors.WithMessage(err, "Assets3d: apiCreateAsset3d: failed to bind json")
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
+			"message": "",
+		})
+		return
+	}
+
+	asset3dID, err := uuid.Parse(c.Param("asset3dID"))
+	if err != nil {
+		err = errors.WithMessage(err, "Assets3d: apiCreateAsset3d: failed to parse asset3dID")
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
+			"message": "",
+		})
+		return
+	}
+
+	crAsset3d, err := a.CreateAsset3d(asset3dID)
+	if err != nil {
+		err = errors.WithMessage(err, "Assets3d: apiCreateAsset3d: failed to create asset3d")
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
+			"message": "",
+		})
+		return
+	}
+
+	err = crAsset3d.SetMeta(in.Meta, false)
+	if err != nil {
+		err = errors.WithMessage(err, "Assets3d: apiCreateAsset3d: failed to set asset3d meta")
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
+			"message": "",
+		})
+		return
+	}
+
+	modFn := func(ops *entry.Asset3dOptions) (*entry.Asset3dOptions, error) {
+		if ops == nil {
+			return nil, errors.New("Assets3d: apiCreateAsset3d: modify func - got nil options")
+		}
+		return ops, nil
+	}
+
+	err = crAsset3d.SetOptions(modFn, false)
+	if err != nil {
+		err = errors.WithMessage(err, "Assets3d: apiCreateAsset3d: failed to set asset3d options")
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
+			"message": "",
+		})
+		return
+	}
+
+	out := crAsset3d.GetEntry()
+	c.JSON(http.StatusOK, out)
```

Comment:
please don't use `entry` package as api response
it's better to define one more type here https://github.com/momentum-xyz/ubercontroller/blob/develop/universe/api/dto/dto.go